AnnService/src/Core/KDT/KDTIndex.cpp

```diff
@@ -148,6 +160,7 @@ namespace SPTAG
             for (DimensionType i = 0; i < m_pGraph.m_iNeighborhoodSize; i++) { \
                 SizeType nn_index = node[i]; \
                 if (nn_index < 0) break; \
+                if (nn_index >= m_pSamples.R()) break; \
```

Comment:
This will not happen if build and refine will not insert VID larger than R. This check will be done thousands of times. We should remove it from critical path.